mmrazor/models/mutables/derived_mutable.py

```diff
@@ -0,0 +1,352 @@
+            `current_mask`. Defaults to None.
+        source_mutables (iterable, optional): Specify source mutables for this
+            derived mutable. If the argument is None, source mutables will be
+            traced automatically by parsing mutables in closure variables.
+            Defaults to None.
+        alias (str, optional): alias of the `MUTABLE`. Defaults to None.
+        init_cfg (dict, optional): initialization configuration dict for
+            ``BaseModule``. OpenMMLab has implement 5 initializer including
+            `Constant`, `Xavier`, `Normal`, `Uniform`, `Kaiming`,
+            and `Pretrained`. Defaults to None.
+
+    Examples:
+        >>> from mmrazor.models.mutables import OneShotMutableChannel
+        >>> mutable_channel = OneShotMutableChannel(
+        ...     num_channels=3,
+        ...     candidate_choices=[1, 2, 3],
+        ...     candidate_mode='number')
+        >>> # derive expand mutable
+        >>> derived_mutable_channel = mutable_channel * 2
+        >>> # source mutables will be traced automatically
+        >>> derived_mutable_channel.source_mutables
+        {OneShotMutableChannel(name=unbind, num_channels=3, current_choice=3, choices=[1, 2, 3], activated_channels=3, concat_mutable_name=[])}  # noqa: E501
+        >>> # modify `current_choice` of `mutable_channel`
+        >>> mutable_channel.current_choice = 2
+        >>> # `current_choice` and `current_mask` of derived mutable will be modified automatically  # noqa: E501
+        >>> derived_mutable_channel
+        DerivedMutable(current_choice=4, activated_channels=4, source_mutables={OneShotMutableChannel(name=unbind, num_channels=3, current_choice=2, choices=[1, 2, 3], activated_channels=2, concat_mutable_name=[])}, is_fixed=False)  # noqa: E501
+    """
+
+    def __init__(self,
+                 choice_fn: Callable,
+                 mask_fn: Optional[Callable] = None,
+                 source_mutables: Optional[Iterable[BaseMutable]] = None,
+                 alias: Optional[str] = None,
+                 init_cfg: Optional[Dict] = None) -> None:
+        super().__init__(alias, init_cfg)
+
+        self.choice_fn = choice_fn
+        self.mask_fn = mask_fn
+
+        if source_mutables is None:
+            source_mutables = self._trace_source_mutables()
+            if len(source_mutables) == 0:
+                # TODO
+                # warning or raise error?
+                logger.warning('Can not find source mutables automatically')
+        else:
+            source_mutables = set(source_mutables)
+        for mutable in source_mutables:
+            if not self.is_source_mutable(mutable):
+                raise ValueError('Expect all mutable to be source mutable, '
+                                 f'but {mutable} is not')
+        self.source_mutables = source_mutables
+
+    # TODO
+    # has no effect
+    def fix_chosen(self, chosen: Dict) -> None:
```

Comment:
Raise Error for `fix_chosen` and `dump_chosen`, when called, as `DerivedMutable` dose not have its own search_space.
Meanwhile, the `DerivedMutable` should be excluded when `export_fix_subnet` and `load_fix_subnet` to imporove human readability.

Reply:
Add warnings when calling `fix_chosen` or `dump_chosen` and add an optional argument `dump_derived_mutable` for function `export_fix_subnet` to indicate whether export derived mutable or not.